source/api/extended.cpp

```diff
@@ -914,6 +915,45 @@ Extended::initiateTransfer(const Models::Address&               inputAddress,
   return bundle.getTransactions();
 }
 
+bool
+Extended::isPromotable(const Types::Trytes& tail) const {
+  if (!Types::isValidHash(tail)) {
+    return false;
+  }
+
+  try {
+    return checkConsistency({ tail }).getState();
+  } catch (IOTA::Errors::BadRequest) {
+    return false;
+  }
+  return false;
+}
+
+Responses::SendTransfer
+Extended::promoteTransaction(const Types::Trytes& tail, int depth, int minWeightMagnitude,
```

Comment:
well, I'm not really a big fan of the interrupt param.
The main reason is that it is completely error prone: because that's taking a const ref,lvalues are accepted, making it possible to calling it with promoteTransaction(....., true|false), meaning it eaither loops undefinitely or does nothing.

I'd rather go for something that prevents that kinds of errors: either discarding the const to only allow rvalue, either not taking any parameter but returning an object that allows you to stop the current process.

If your concern is to spawn a thread in that function, an alternative design could be to take a lambda that returns true or false. The lambda is evaluated at each iteration and leaves it possible for the caller to implement asynchronous or synchronous operations on top of it.

Reply:
I totally agree, the interrupt bool has been replaced with a std::function allowing the using to passe a stateful lambda. Thus, he can parametrize the interruption of the method as he wants : number of loops, timer...